### PR TITLE
fix: eagle3 quantized base model

### PR DESCRIPTION
**Pull request by h-guo18**
## What does this PR do?

**Type of change:** Bug Fix 

**Overview:** 

Fixed 2 bugs when training with quantized base model:
- Some base model modules are not loaded but present in _keep_in_fp32_modules , raising an error [here](https://github.com/huggingface/transformers/blob/v4.56.2/src/transformers/modeling_utils.py#L2248) when loading model
  - Solution: Instead of setting `num_hidden_layer=0` to `from_pretrained()` to load partial base model, we load the entire base model to cpu first, then delete `base_model.layers` during `convert()`.
  
- The base model is considered "purely quantized models" in HF trainer, therefore cannot be fine-tuned. Error raised [here](https://github.com/huggingface/transformers/blob/v4.56-release/src/transformers/trainer.py#L566) when initializing trainer.
  - Solution: setting `base_model.is_quantized=False` during model conversion; 

Other minor polishes/fix:
- Move `ARValidationCallback` to utils for better clarity of `main.py`;
- Removed redundant `trainer._move_model_to_device()` in `main.py` since it will be called during trainer intialization and `trainer.train()`.
  - ref: https://github.com/huggingface/transformers/blob/v4.56-release/src/transformers/trainer.py#L620 
  - ref: https://github.com/huggingface/transformers/blob/v4.56-release/src/transformers/trainer.py#L2267

- Use `sdpa` attention during `pseudo_generate` instead of `flex_attn` to avoid compile error and improve performance.
 

## Usage
Unchanged

```python
# Add a code snippet demonstrating how to use this
```

## Testing
Tested with dummy training on both onlin/offline and quantized/unquantized settings:
- TinyLlama, online
- TinyLlama, offline
- GPTOSS-20b, online
- GPTOSS-120b, offline

Saw training, evaluation and checkpoint saving with not error, and loss decreasing.

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No

### CODE REVIEW

File: examples/speculative_decoding/eagle_utils.py [hunk 1]
```diff
@@ -19,11 +19,21 @@
...
 import transformers
+from ar_validate import validate_ar
+from datasets import load_dataset
 from torch.utils.data import Dataset
+from transformers import TrainerCallback
 from transformers.trainer_pt_utils import LabelSmoother
 
 from modelopt.torch.utils import print_rank_0
 
+try:
```
Comment (yeyu-nvidia) [inline examples/speculative_decoding/eagle_utils.py:30]: why do we move this from main to here? do you only want to use wandb for eagle? what about in the future we have some other algorithms?

File: modelopt/torch/speculative/plugins/transformers.py [hunk 2]
```diff
@@ -445,12 +445,20 @@ def modify(
...
         # EAGLE-3 auxiliary hidden_states
-        if self.eagle_config.use_aux_hidden_state:
+        if (not eagle_offline) and self.eagle_config.use_aux_hidden_state:
             self._aux_hidden_states = []
             for layer_idx, layer in enumerate(self.model.layers):
                 if layer_idx in self.eagle_config.eagle_aux_hidden_state_layer_ids:
                     layer.register_forward_hook(self._collect_aux_hidden_states_forward_hook)
 
+        # delete base model layers for offline training
+        if eagle_offline:
```
Comment (yeyu-nvidia) [inline modelopt/torch/speculative/plugins/transformers.py:455]: This may lead to CUDA OOM if the model is too big to fit in the GPU. I think the previous way (model_kwargs = {"num_hidden_layers": 0}) has a point that user can use a small GPU to train offline eagle. Can we follow that and use the hack to avoid the post_init check?

Comment (h-guo18) [reply modelopt/torch/speculative/plugins/transformers.py:455]: Model conversion is currently done in CPU memory, so as long as the whole base model fit in CPU memory, it should be fine.

The problem with `_keep_in_fp32_modules` is that, it is read from base model class definition (see ref link below), so I did not find a good place to manipulate it. Is there a good way?

ref: 
- https://github.com/huggingface/transformers/blob/v4.56-release/src/transformers/models/gpt_oss/modeling_gpt_oss.py#L404

- https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L2055
